File: rb/Irrgarten/GameState.rb
#encoding:utf-8

module Irrgarten
    class GameState

        @labyrinth
        @players
        @monsters
        @currentPlayer
        @winner
        @log

        def initialize(labyrinth, players, monsters, currentPlayer, winner, log)
            @labyrinth = labyrinth
            @players = players
            @monsters = monsters
            @currentPlayer = currentPlayer
            @winner = winner
            @log = log
        end

        def get_labyrinth
            return @labyrinth
        end

        def get_players
            return @players
        end

        def get_monsters
            return @monsters
        end

        def get_current_player
            return @currentPlayer
        end

        def get_winner
            return @winner
        end

        def get_log
            return @log
        end
    end
end

File: ExamenP5/Irrgarten/src/irrgarten/GameState.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author Sergio Salvador Gil
 */
public class GameState {
    private String labyrinth;
    private String players;
    private String monsters;
    private int currentPlayer;
    private int bearerOfCataclysm;
    private boolean winner;
    private String log;
    
    GameState(String laberinto, String jugadores, String monstruos, int actual,
            boolean ganador, String registro, int portador){
        labyrinth=laberinto;
        players=jugadores;
        monsters=monstruos;
        currentPlayer=actual;
        winner=ganador;
        log=registro;
        bearerOfCataclysm=portador;
    }
    
    public String getLabyrinth(){
        return labyrinth;
    }
    public String getPlayers(){
        return players;
    }
    public String getMonsters(){
        return monsters;
    }
    public int getCurrentPlayer(){
        return currentPlayer;
    }
    public int getBearer(){
        return bearerOfCataclysm;
    }
    public boolean isWinner(){
        return winner;
    }
    public String getLog(){
        return log;
    }
}


